masonry_core/README.md

```diff
@@ -1,3 +1,75 @@
-# masonry_core
+# Masonry Core
+
+<div align="center">
+
+**Traits and types of the Masonry toolkit**
+
+[![Latest published version.](https://img.shields.io/crates/v/masonry_core.svg)](https://crates.io/crates/masonry_core)
+[![Documentation build status.](https://img.shields.io/docsrs/masonry_core.svg)](https://docs.rs/masonry_core)
+[![Apache 2.0 license.](https://img.shields.io/badge/license-Apache--2.0-blue.svg)](#license)
+\
+[![Linebender Zulip chat.](https://img.shields.io/badge/Linebender-%23masonry-blue?logo=Zulip)](https://xi.zulipchat.com/#narrow/channel/317477-masonry)
+[![GitHub Actions CI status.](https://img.shields.io/github/actions/workflow/status/linebender/xilem/ci.yml?logo=github&label=CI)](https://github.com/linebender/xilem/actions)
+[![Dependency staleness status.](https://deps.rs/crate/masonry_core/latest/status.svg)](https://deps.rs/crate/masonry_core)
+
+</div>
+
+<!-- We use cargo-rdme to update the README with the contents of lib.rs.
+To edit the following section, update it in lib.rs, then run:
+cargo rdme --workspace-project=masonry_core --heading-base-level=0
+Full documentation at https://github.com/orium/cargo-rdme -->
+
+<!-- Intra-doc links used in lib.rs should be evaluated here.
+See https://linebender.org/blog/doc-include/ for related discussion. -->
+
+
+<!-- cargo-rdme start -->
 
 Traits and types of the Masonry toolkit.
+See [Masonry's documentation] for more details, examples and resources.
+
+[Masonry's documentation]: https://docs.rs/masonry/latest/
+
+<!-- cargo-rdme end -->
+
+## Minimum supported Rust Version (MSRV)
+
+This version of Masonry Core has been verified to compile with **Rust 1.88** and later.
+
+Future versions of Masonry Core might increase the Rust version requirement.
+It will not be treated as a breaking change and as such can even happen with small patch releases.
+
+<details>
+<summary>Click here if compiling fails.</summary>
+
+As time has passed, some of Masonry Core's dependencies could have released versions with a higher Rust requirement.
+If you encounter a compilation issue due to a dependency and don't want to upgrade your Rust toolchain, then you could downgrade the dependency.
+
+```sh
+# Use the problematic dependency's name and version
+cargo update -p package_name --precise 0.1.1
+```
+
+</details>
+
+<!-- We hide these elements when viewing in Rustdoc, because they're not expected to be present in crate level docs -->
+<div class="rustdoc-hidden">
```

Comment:
Nit: This is no longer needed since  we use cargo rdme. If this is still present in e.g. Xilem Core, we should remove it there as well